`GradleTest.testJacksonBomDependency` fails for `jackson-annotations` 2.21-SNAPSHOT dependency

Looks like there is a new failure for `GradleTest`, probably related to versioning change for `jackson-annotations`:

```
[ERROR] Failures: 
[ERROR]   GradleTest.testJacksonBomDependency:33->build:61 FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':checkMetadata'.
> Missing dependency to 'jackson-bom'. Gradle Metadata publishing is most likely broken:
    - com.fasterxml.jackson.core:jackson-annotations:2.21-SNAPSHOT
```

I tried adding 

```
 val modulesWithoutGradleMetadata = listOf(
         "com.fasterxml.jackson.jr:jackson-jr-all", // TODO is there a reason not to add this?
 
+        "com.fasterxml.jackson:jackson-annotations", // due to 2.x/3.x co-constraints
         "com.fasterxml.jackson:jackson-bom", // does not need it
         "com.fasterxml.jackson.module:jackson-module-scala_2.11", // built with sbt
         "com.fasterxml.jackson.module:jackson-module-scala_2.12", // built with sbt
```

in `src/test/resources/com/fasterxml/jackson/integtest/gradle/build.gradle.kts` but that did not prevent failure.

This is for `2.x` branch (default)

--- a/src/test/resources/com/fasterxml/jackson/integtest/gradle/build.gradle.kts
+++ b/src/test/resources/com/fasterxml/jackson/integtest/gradle/build.gradle.kts
@@ -5,6 +5,8 @@ plugins {
 val modulesWithoutGradleMetadata = listOf(
         "com.fasterxml.jackson.jr:jackson-jr-all", // TODO is there a reason not to add this?
 
+        "com.fasterxml.jackson.core:jackson-annotations", // due to 2.x/3.x co-constraints
+
         "com.fasterxml.jackson:jackson-bom", // does not need it
         "com.fasterxml.jackson.module:jackson-module-scala_2.11", // built with sbt
         "com.fasterxml.jackson.module:jackson-module-scala_2.12", // built with sbt

